controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/CloudEventReporter.java

```diff
@@ -27,33 +32,67 @@
 
     private final IssueHandler issueHandler;
     private final AwsEventFetcher eventFetcher;
-    private final Set<String> awsRegions;
+    private final Map<String, List<ZoneApi>> zonesByCloudNativeRegion;
+    private final NodeRepository nodeRepository;
 
     CloudEventReporter(Controller controller, Duration interval, JobControl jobControl) {
         super(controller, interval, jobControl);
         this.issueHandler = controller.serviceRegistry().issueHandler();
         this.eventFetcher = controller.serviceRegistry().eventFetcherService();
-        this.awsRegions = controller.zoneRegistry().zones()
-                .ofCloud(CloudName.from("aws"))
-                .reachable()
-                .zones().stream()
-                .map(ZoneApi::getCloudNativeRegionName)
-                .collect(Collectors.toSet());
+        this.nodeRepository = controller.serviceRegistry().configServer().nodeRepository();
+        this.zonesByCloudNativeRegion = getZonesByCloudNativeRegion();
     }
 
     @Override
     protected void maintain() {
         log.log(Level.INFO, "Fetching events for cloud hosts.");
-        for (var awsRegion : awsRegions) {
+        for (var awsRegion : zonesByCloudNativeRegion.keySet()) {
             List<CloudEvent> events = eventFetcher.getEvents(awsRegion);
             for (var event : events) {
-                Issue issue = eventFetcher.createIssue(event);
-                if (!issueHandler.issueExists(issue)) {
-                    issueHandler.file(issue);
-                    log.log(Level.INFO, String.format("Filed an issue with the title '%s'", issue.summary()));
-                }
+                deprovisionHosts(awsRegion, event);
+                submitIssue(event);
             }
         }
     }
 
+    private void deprovisionHosts(String awsRegion, CloudEvent event) {
+        for (var zone : zonesByCloudNativeRegion.get(awsRegion)) {
+            nodeRepository.list(zone.getId())
+                    .stream()
+                    .filter(shouldDeprovisionHost(event))
+                    .map(node -> node.hostname().value())
+                    .forEach(hostname -> {
+                        log.info(String.format("Setting host %s to wantToRetire and wantToDeprovision", hostname));
+                        nodeRepository.retireAndDeprovision(zone.getId(), hostname);
+                        event.affectedInstances.removeIf(hostname::contains);
+                    });
+        }
+    }
+
+    private void submitIssue(CloudEvent event) {
+        if (event.affectedInstances.isEmpty())
+            return;
+        Issue issue = eventFetcher.createIssue(event);
+        if (!issueHandler.issueExists(issue)) {
+            issueHandler.file(issue);
+            log.log(Level.INFO, String.format("Filed an issue with the title '%s'", issue.summary()));
+        }
+    }
+
+    private Predicate<Node> shouldDeprovisionHost(CloudEvent event) {
```

Comment:
Also check if the node already has `wantToDeprovision` and `wantToRetire` set already.